validator/client/validator_duties.go

```diff
@@ -0,0 +1,121 @@
+package client
+
+import (
+	"context"
+	"io"
+	"time"
+
+	"github.com/pkg/errors"
+	ethpb "github.com/prysmaticlabs/ethereumapis/eth/v1alpha1"
+	"github.com/prysmaticlabs/prysm/shared/bytesutil"
+	"github.com/prysmaticlabs/prysm/shared/slotutil"
+	"go.opencensus.io/trace"
+)
+
+// StreamDuties consumes a server-side stream of validator duties from a beacon node
+// for a set of validating keys passed in as a request type. New duties will be
+// sent over the stream upon a new epoch being reached or from a a chain reorg happening
+// across epochs in the beacon node.
+func (v *validator) StreamDuties(ctx context.Context) error {
+	ctx, span := trace.StartSpan(ctx, "validator.StreamDuties")
+	defer span.End()
+
+	validatingKeys, err := v.keyManager.FetchValidatingKeys()
+	if err != nil {
+		return err
+	}
+	req := &ethpb.DutiesRequest{
+		PublicKeys: bytesutil.FromBytes48Array(validatingKeys),
+	}
+	stream, err := v.validatorClient.StreamDuties(ctx, req)
+	if err != nil {
+		return errors.Wrap(err, "Could not setup validator duties streaming client")
+	}
+	for {
+		res, err := stream.Recv()
+		// If the stream is closed, we stop the loop.
+		if err == io.EOF {
+			break
+		}
+		// If context is canceled we stop the loop.
+		if ctx.Err() == context.Canceled {
+			return errors.Wrap(ctx.Err(), "context has been canceled so shutting down the loop")
+		}
+		if err != nil {
+			return errors.Wrap(err, "could not receive validator duties from stream")
```

Comment:
Likewise here, we should probably just log an error instead of closing this function?